cartography/intel/okta/test.py

```diff
@@ -0,0 +1,796 @@
+    SET org.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(ingest,
+                      ORG_NAME=organization,
+                      okta_update_tag=okta_update_tag)
+
+
+def _load_okta_groups(neo4j_session, okta_org_id, group_list, okta_update_tag):
+    ingest_statement = """
+    MATCH (org:OktaOrganization{id: {ORG_ID}})
+    WITH org
+    UNWIND {GROUP_LIST} as group_data
+    MERGE (new_group:OktaGroup{id: group_data.id})
+    ON CREATE SET new_group.firstseen = timestamp()
+    SET new_group.name = group_data.name,
+    new_group.description = group_data.description,
+    new_group.sam_account_name = group_data.sam_account_name,
+    new_group.dn = group_data.dn,
+    new_group.windows_domain_qualified_name = group_data.windows_domain_qualified_name,
+    new_group.external_id = group_data.external_id,
+    new_group.lastupdated = {okta_update_tag}
+    WITH new_group, org
+    MERGE (org)-[org_r:RESOURCE]->(new_user)
+    ON CREATE SET org_r.firstseen = timestamp()
+    SET org_r.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(ingest_statement,
+                      ORG_ID=okta_org_id,
+                      GROUP_LIST=group_list,
+                      okta_update_tag=okta_update_tag)
+
+
+def _get_okta_groups_id_from_graph(neo4j_session, okta_org_id):
+    group_query = "MATCH (:OktaOrganization{id: {ORG_ID}})-[:RESOURCE]->(group:OktaGroup) return group.id as id"
+
+    result = neo4j_session.run(group_query, ORG_ID=okta_org_id)
+
+    groups = [r['id'] for r in result]
+
+    return groups
+
+
+def _get_okta_application_id_from_graph(neo4j_session, okta_org_id):
+    app_query = "MATCH (:OktaOrganization{id: {ORG_ID}})-[:RESOURCE]->(app:OktaApplication) return app.id as id"
+
+    result = neo4j_session.run(group_query, ORG_ID=okta_org_id)
+
+    apps = [r['id'] for r in result]
+
+    return apps
+
+
+def _get_okta_group_members(api_client, group_id):
+    member_list = []
+    next_url = None
+
+    while True:
+        try:
+            # https://developer.okta.com/docs/reference/api/groups/#list-group-members
+            if next_url:
+                paged_response = api_client.get(next_url)
+            else:
+                params = {
+                    'limit': 1000
+                }
+                paged_response = api_client.get_path('{0}/users'.format(group_id), params)
+        except OktaError as okta_error:
+            break
+
+        member_results = json.loads(paged_response.text)
+
+        for member in member_results:
+            member_list.append(member.id)
+
+        if not _is_last_page(paged_response):
+            next_url = paged_response.links.get("next").get("url")
+        else:
+            break
+
+    return member_list
+
+
+def _load_okta_group_membership(neo4j_session, api_client, okta_org_id, okta_update_tag):
+
+    for group_id in _get_okta_groups_id_from_graph(neo4j_session, okta_org_id):
```

Comment:
Instead of re-querying the graph and re-processing the data to a python object again, I recommend using something like our [_attach pattern](https://github.com/lyft/cartography/blob/master/cartography/intel/aws/rds.py#L108).  

In this case, our _attach function would match on our existing Okta groups and then merge in the `members`.